x/evm/state/balance.go

```diff
@@ -0,0 +1,103 @@
+package state
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/sei-protocol/sei-chain/x/evm/types"
+)
+
+func (s *StateDBImpl) CreateAccount(common.Address) {
+	// noop
+	// EVM account creation is handled in ante handlers.
+	// State initialization is handled in Get/SetState
+}
+
+func (s *StateDBImpl) SubBalance(evmAddr common.Address, amt *big.Int) {
+	if amt.Sign() == 0 {
+		return
+	}
+	if amt.Sign() < 0 {
+		s.AddBalance(evmAddr, new(big.Int).Neg(amt))
+		return
+	}
```

Comment:
out of curiosity, why do we want to support sub/add'ing negative amounts?

Reply:
these functions are called by go-ethereum, and the original implementation makes no assumption on the sign of `amt`, so just want to make sure we support the same behavior